pkg/provider/passworddepot/passworddepot_api.go

```diff
@@ -0,0 +1,344 @@
+	Login       string    `json:"login"`
+	Pass        string    `json:"pass"`
+	URL         string    `json:"url"`
+	Importance  string    `json:"importance"`
+	Date        time.Time `json:"date"`
+	Comment     string    `json:"comment"`
+	Expirydate  string    `json:"expirydate"`
+	Tags        string    `json:"tags"`
+	Author      string    `json:"author"`
+	Category    string    `json:"category"`
+	Icon        string    `json:"icon"`
+	Secondeye   string    `json:"secondeye"`
+	Secondpass  string    `json:"secondpass"`
+	Template    string    `json:"template"`
+	Acm         string    `json:"acm"`
+	Paramstr    string    `json:"paramstr"`
+	Loginid     string    `json:"loginid"`
+	Passid      string    `json:"passid"`
+	Donotaddon  string    `json:"donotaddon"`
+	Markassafe  string    `json:"markassafe"`
+	Safemode    string    `json:"safemode"`
+}
+
+var errDBNotFound = errors.New("Database not found")
+var errSecretNotFound = errors.New("Secret not found")
+
+// load tls certificates
+
+func NewAPI(baseURL, username, password, hostPort string) (*API, error) {
+	api := &API{
+		baseURL:  baseURL,
+		hostPort: hostPort,
+		username: username,
+		password: password,
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{MinVersion: tls.VersionTLS12},
+	}
+
+	api.client = &http.Client{Transport: tr}
+	err := api.login()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to login")
+	}
+	return api, nil
+}
+
+func (api *API) doAuthenticatedRequest(r *http.Request) (*http.Response, error) {
+	r.Header.Add("access_token", api.secret.AccessToken)
+	r.Header.Add("client_id", api.secret.ClientID)
+
+	return api.client.Do(r)
+}
+func (api *API) getDatabaseFingerprint(database string) (string, error) {
+	databases, err := api.ListDatabases()
+	if err != nil {
+		return "", errors.Wrap(err, "error: getting database list")
+	}
+
+	for _, db := range databases.Databases {
+		if strings.Contains(db.Name, database) {
+			return db.Fingerprint, nil
+		}
+	}
+
+	return "", errDBNotFound
+}
+
+func (api *API) getSecretFingerprint(databaseFingerprint, secretName, folder string) (string, error) {
+	secrets, err := api.ListSecrets(databaseFingerprint, folder)
+	if err != nil {
+		return "", errors.Wrap(err, "error: getting secrets list")
+	}
+
+	parts := strings.Split(secretName, ".")
+	searchName := parts[0]
+	var fingerprint string
+	for _, entry := range secrets.Entries {
+		if strings.Contains(entry.Name, searchName) {
+			fingerprint = entry.Fingerprint
+			if len(parts) > 1 {
+				return api.getSecretFingerprint(databaseFingerprint, strings.Join(parts[1:], "."), fingerprint)
+			}
+			return fingerprint, nil
+		}
+	}
+	return "", errSecretNotFound
+}
+
+func (api *API) getendpointURL(endpoint string) string {
+	return fmt.Sprintf("https://%s:%s/v1.0/%s", api.baseURL, api.hostPort, endpoint)
+}
+
+func (api *API) login() error {
+	loginRequest, err := http.NewRequest("GET", api.getendpointURL("login"), http.NoBody)
```

Comment:
These requests should have a context with a timeout.

Reply:
Added context.